src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -436,18 +457,63 @@ final class LazyList[+A] private(private[this] var lazyState: () => LazyList.Sta
   private def zipLinearSeq[B](that: collection.LinearSeq[B]): LazyList[(A, B)] =
     newLL {
       if (this.isEmpty || that.isEmpty) State.Empty
-      else sCons((this.head, that.head), tail.zipLinearSeq(that.tail))
+      else sCons((this.head, that.head), this.tail zipLinearSeq that.tail)
     }
 
   private def zipState[B](it: Iterator[B]): State[(A, B)] =
     if (this.isEmpty || !it.hasNext) State.Empty
     else {
       lazy val evaluatedElem = it.next()
-      sCons((head, evaluatedElem), newLL { evaluatedElem; tail.zipState(it) })
+      sCons((head, evaluatedElem), newLL { evaluatedElem; tail zipState it })
     }
 
   override def zipWithIndex: LazyList[(A, Int)] = this.zip(LazyList.from(0))
 
+  override def zipAll[A1 >: A, B](that: scala.Iterable[B], thisElem: A1, thatElem: B): LazyList[(A1, B)] = {
+    if (this.knownIsEmpty) {
+      if (that.knownSize == 0) LazyList.empty
+      else LazyList.continually(thisElem) zip that
+    } else {
+      if (that.knownSize == 0) zip(LazyList.continually(thatElem))
+      else that match {
+        case that: collection.LinearSeq[B] => zipAllLinearSeq(that, thisElem, thatElem)
+        case _                             => newLL(zipAllState(that.iterator, thisElem, thatElem))
+      }
+    }
+  }
+
+  private def zipAllLinearSeq[A1 >: A, B](that: collection.LinearSeq[B], thisElem: A1, thatElem: B): LazyList[(A1, B)] =
+    newLL {
+      if (this.isEmpty) {
+        if (that.isEmpty) State.Empty
+        else sCons((thisElem, that.head), LazyList.continually(thisElem) zipLinearSeq that.tail)
+      } else {
+        if (that.isEmpty) sCons((this.head, thatElem), this.tail zipLinearSeq LazyList.continually(thatElem))
+        else sCons((this.head, that.head), this.tail.zipAllLinearSeq(that.tail, thisElem, thatElem))
+      }
+    }
+
+  private def zipAllState[A1 >: A, B](it: Iterator[B], thisElem: A1, thatElem: B): State[(A1, B)] = {
+    if (it.hasNext) {
+      lazy val evaluatedElem = it.next()
+      if (this.isEmpty) sCons((thisElem, evaluatedElem), newLL { evaluatedElem; LazyList.continually(thisElem) zipState it })
+      else sCons((this.head, evaluatedElem), newLL { evaluatedElem; this.tail.zipAllState(it, thisElem, thatElem) })
+    } else {
+      if (this.isEmpty) State.Empty
+      else sCons((this.head, thatElem), this.tail zipLinearSeq LazyList.continually(thatElem))
+    }
+  }
+
+  // just in case it can be meaningfully overridden at some point
+  override def lazyZip[B](that: scala.Iterable[B]): LazyZip2[A, B, LazyList.this.type] =
+    super.lazyZip(that)
```

Comment:
I missed that `lazyZip` has become a proper method. When did it happen?

Reply:
I have no idea. It's not really useful on `LazyList`, whose `zip` is already lazy

Reply:
https://github.com/scala/collection-strawman/pull/223

Reply:
FTR, I left this comment but it seems that it was ignored: https://github.com/scala/bug/issues/11063#issuecomment-411846278 (@szeiger ?)